### ISSUE TITLE
Can't open jp2 files exported with 3.8.0 on Ubuntu 20.04

### ISSUE THREAD

**Issue by faf2019** [darktable-org/darktable#10749]
Since version 3.8.0, jp2 files exported from Darktable can no longer be opened with Gimp and with Darktable itself.

I installed Darktable from the packages at https://launchpad.net/~ubuntuhandbook1/+archive/ubuntu/darktable

![image](https://user-images.githubusercontent.com/32289337/147787365-e3e79aa9-6c11-4641-ba4c-4da17a331ffe.png)
![image](https://user-images.githubusercontent.com/32289337/147787442-2cc8ebc9-f710-404c-98a5-b032b6d9f1ef.png)

```
this is darktable 3.8.0
copyright (c) 2009-2021 johannes hanika
darktable-dev@lists.darktable.org

compile options:
  bit depth is 64 bit
  normal build
  SSE2 optimized codepath enabled
  OpenMP support enabled
  OpenCL support enabled
  Lua support disabled
  Colord support enabled
  gPhoto2 support enabled
  GraphicsMagick support enabled
  ImageMagick support disabled
  OpenEXR support enabled
```

**Comment by TurboGit** [comment]
Just tested and I was able to open a j2k generated by dt with GIMP 2.10.30.

**Comment by TurboGit** [comment]
What are the export options used?

**Comment by faf2019** [comment]
format: jp2
profile : Adobe rvb

GIMP version 2.10.18 (Ubuntu 20.04 version)

![image](https://user-images.githubusercontent.com/32289337/147790574-b1ec6f9e-66ca-42d2-b8d7-641eaa9f9f3a.png)

**Comment by faf2019** [comment]
Here is a small jp2 sample : https://fromsmash.com/jp2sample

**Comment by MStraeten** [comment]
reproducible with my osx build. gimp error message indicates a wrong header.

**Comment by TurboGit** [comment]
I can see that with your jp2, now we need to understand what I do differently. I'm thinking maybe you are exporting with tags & xmp integrated? Can you open the preferences in the export module (hamburger top right) and deselect all options?

**Comment by faf2019** [comment]
I have the same result by unchecking all the boxes

![image](https://user-images.githubusercontent.com/32289337/147813091-f728aeca-fc43-4007-b135-e7416b2a3c51.png)

**Comment by TurboGit** [comment]
Ok, and using the export in pixels instead of `inch (for printing)`?

**Comment by TurboGit** [comment]
This is really strange, I have tested lot of combination and was always able to open in GIMP!

**Comment by faf2019** [comment]
same result using the export in pixels.

![image](https://user-images.githubusercontent.com/32289337/147813545-7234ba77-5838-4152-bbcd-573c8960ee38.png)

Yet I didn't change my version of Gimp, I just updated Darktable.

**Comment by TurboGit** [comment]
Which version of jp2 library are you using, I have 2.4.0:

```
libopenjp2-7-dev:
  Installed: 2.4.0-3
  Candidate: 2.4.0-3
```

**Comment by TurboGit** [comment]
Side question, do you have this with all images? Even one that was not edited at all (just default modules)?

**Comment by faf2019** [comment]
> Which version of jp2 library are you using, I have 2.4.0:

dpkg -s libopenjp2-7

Package: libopenjp2-7
Status: install ok installed
Priority: optional
Section: libs
Installed-Size: 399
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Multi-Arch: same
Source: openjpeg2
Version: 2.3.1-1ubuntu4.20.04.1

**Comment by faf2019** [comment]
> Side question, do you have this with all images? Even one that was not edited at all (just default modules)?

I have the problem with all images, even when using no module.

**Comment by TurboGit** [comment]
So maybe a bug in lib 2.3.1 fixed in 2.4.0. @MStraeten : Which version of lib jp2 do you have on MacOS?

**Comment by MStraeten** [comment]
```
-rwxr-xr-x  1 root  admin  321096 29 Okt 01:20 /opt/local/lib/libopenjp2.2.4.0.dylib
lrwxr-xr-x  1 root  admin      22 29 Okt 01:20 /opt/local/lib/libopenjp2.7.dylib -> libopenjp2.2.4.0.dylib
-rw-r--r--  1 root  admin  362448 29 Okt 01:20 /opt/local/lib/libopenjp2.a
lrwxr-xr-x  1 root  admin      18 29 Okt 01:20 /opt/local/lib/libopenjp2.dylib -> libopenjp2.7.dylib
```
